packages/spectacle/src/components/slide-layout.tsx

```diff
@@ -195,11 +196,124 @@ const Quote = ({
     </Box>
   </Slide>
 );
+
+/**
+ * Generic Codepane utility with optional Description text
+ */
+const CodeLayout = ({
+  children,
+  language,
+  text,
+  textProps,
+  ...props
+}: Omit<SlideProps, 'children'> & {
+  children: string;
+  language: string | undefined;
+  text?: string | ReactNode;
+  textProps?: ComponentProps<typeof Text>;
+}) => (
+  <Box data-testid="CodePane">
+    {text ? (
+      <Text margin={8} {...textProps}>
+        {text}
+      </Text>
+    ) : null}
+    <CodePane language={language} {...props}>
+      {children}
+    </CodePane>
+  </Box>
+);
+
+/**
+ * single Code Pane with optional Title layout
+ */
+const Code = ({
+  children,
+  language,
+  title,
+  titleProps,
+  codePaneProps,
+  ...rest
+}: Omit<SlideProps, 'children'> & {
+  children: string;
+  language: string;
+  title?: string | ReactNode;
+  titleProps?: ComponentProps<typeof Text>;
+  codePaneProps?: CodePaneProps;
+}) => {
+  return (
+    <Slide {...rest}>
+      <Box display="inline-block" style={{ overflow: 'scroll' }}>
+        {title ? <Heading {...titleProps}>{title}</Heading> : null}
+        <CodeLayout language={language} {...codePaneProps}>
+          {children}
+        </CodeLayout>
+      </Box>
+    </Slide>
+  );
+};
+
+interface CodeBlocks extends CodePaneProps {
```

Comment:
Here's what I came up with to deal with `CodePaneProps` from our [previous discussion about CodePaneProps](https://github.com/FormidableLabs/spectacle/pull/1217#discussion_r961613775).

I would have liked to extend the CodePaneProps (inline down here)[https://github.com/FormidableLabs/spectacle/pull/1217/files#diff-c1ac090b2ff636b65120246c862260a07e29ceacc75778bc3c88ec0d89868a9bR270] but how to do that gracefully was lost on me. @gksander, do you see a more elegant way of doing this? Thanks!